core/object/class_db.cpp

```diff
@@ -165,6 +166,10 @@ uint64_t ClassDB::get_api_hash(APIType p_api) {
 	OBJTYPE_RLOCK;
 #ifdef DEBUG_METHODS_ENABLED
 
+	if (api_hashes_cache.has(p_api)) {
+		return api_hashes_cache[p_api];
+	}
```

Comment:
Can the API be updated after this cache has been filled? E.g. using the ClassDB API from script / GDExtension to add new classes/methods/properties at runtime?

Reply:
From that point on, the current API type is set to `API_TYPE_NONE`, signaling that the core and the editor APIs are already fully fed. Extensions, upon being discovered and loaded, can contribute to the class DB, but their classes are registered as `API_TYPE_EXTENSION` or `API_TYPE_EDITOR_EXTENSION`, so they can be safely ignored.

Reply:
Even if I believe extension API is not hashed currently, I had your concerns in the back of my mind, and felt better by doing this one: #76467.